change/@ni-nimble-blazor-0d8d8520-2d1c-4360-b924-13d1829a35eb.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Is it possible to write theory tests for this?
Use ClassData to return a component reference that implements the interface, IGroupable, etc, and then test those attributes?

Seems like this approach isn't making things easier to track, there are a lot more failure points, i.e. adding a new method to the base class doesn't guarantee it's called, you can call the wrong base class method, it's more verbose, not less

Reply:
It looks like Xunit supports `[Fact]` on base classes, so this seems to work and is less verbose:
![XunitTestsInBaseClass](https://user-images.githubusercontent.com/20709258/236042349-55cb6a40-89c2-4f54-b1cb-ac3f9246c03a.PNG)
![XunitTestsInBaseClass2](https://user-images.githubusercontent.com/20709258/236042673-b0c08e3e-eb16-4594-8714-efffb52b61ac.PNG)

So that's an option too if people are fine with that approach (I couldn't find any guidance either way about this in the test convention wiki docs).

Reply:
That's great. I've removed the test functions from the derived classes and am just letting the base classes provide the tests. I think this addresses Milan's concerns about more failure points and verbosity.